src/ApiException.php

```diff
@@ -78,6 +78,50 @@ public function getStatus()
         return $this->status;
     }
 
+    // Returns null if metadata does not contain error info, or return containsErrorInfo() array
+    // if the metadata does contain error info.
+    public static function decodeMetadataErrorInfo($metadata)
+    {
+        // For the createFromRpcStatus() method, $metadata is an object.
+        if (is_object($metadata)) {
+            $metadataRpcStatus = Serializer::decodeAnyMessages($metadata);
+            $detailsRpcStatus = self::containsErrorInfo($metadataRpcStatus);
+            if ($detailsRpcStatus['containsErrorInfo']) {
+                return  $detailsRpcStatus;
+            }
+        } elseif (is_Array($metadata)) {
+            // For REST-based responses, the $metadata does not need to be decoded.
+            $detailsRest = self::containsErrorInfo($metadata);
+            if ($detailsRest['containsErrorInfo']) {
+                return $detailsRest;
+            }
+        }
+        $metadataGrpc = Serializer::decodeMetadata($metadata);
+        $detailsGrpc = self::containsErrorInfo($metadataGrpc);
+        return $detailsGrpc['containsErrorInfo'] ? $detailsGrpc : null;
```

Comment:
```suggestion
        $details = [];
        if (is_object($metadata)) {
            $metadataRpcStatus = Serializer::decodeAnyMessages($metadata);
            $details = self::containsErrorInfo($metadataRpcStatus);
        } elseif (is_Array($metadata)) {
            // For REST-based responses, the $metadata does not need to be decoded.
            $details = self::containsErrorInfo($metadata);
        } else {
            $metadataGrpc = Serializer::decodeMetadata($metadata);
            $details = self::containsErrorInfo($metadataGrpc);
        }
       
        return $details['containsErrorInfo'] ? $details : null;
```

Reply:
Don't feel like you need to accept the suggestion, you can just copy-paste it if you want to tweak/try it

Reply:
Thanks for the suggestion! I took it and it helped me fix up the function such that all the tests pass, so looks like the original way it was working was creating the seg fault error.